src/battle_util.c

```diff
@@ -9060,7 +9060,10 @@ static u32 CalcFinalDmg(u32 dmg, u16 move, u8 battlerAtk, u8 battlerDef, u8 move
     }
 
     // check stab
-    if (IS_BATTLER_OF_TYPE(battlerAtk, moveType) && move != MOVE_STRUGGLE && move != MOVE_NONE)
+    // Pokemon will get STAB after type change, so include in the AI decision dmg calcs
+    if ((IS_BATTLER_OF_TYPE(battlerAtk, moveType) || 
+        ((abilityAtk == ABILITY_PROTEAN || abilityAtk == ABILITY_LIBERO) && !IsNeutralizingGasOnField()))
```

Comment:
the `IsNeutralizingGasOnField` check is redundant, since it's included in `GetBattlerAbility`